springboard_iogt/views.py

```diff
@@ -3,11 +3,37 @@
 from itertools import chain
 
 from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound, HTTPNotFound
 from elasticutils import F
 
 from springboard.views.base import SpringboardViews
 
-from springboard_iogt.utils import randomize_query
+from springboard_iogt.utils import (
+    randomize_query, get_redirect_url, get_matching_route)
+
+
+ONE_YEAR = 31536000
+
+PERSONAE = {'CHILD', 'TEENAGER', 'PARENT', 'WORKER'}
+PERSONA_COOKIE_NAME = 'iogt-persona'
+PERSONA_SKIP_COOKIE_VALUE = '__skip__'
+PERSONA_REDIRECT_ROUTES = {'home', 'page', 'category', 'flat_page'}
+
+
+def persona_tween_factory(handler, registry):
+
+    def persona_tween(request):
+        if PERSONA_COOKIE_NAME in request.cookies:
+            return handler(request)
+
+        route = get_matching_route(request)
+        if route and route.name in PERSONA_REDIRECT_ROUTES:
```

Comment:
`get_matching_route` returns None if it doesn't match. So it will end up doing `return handler(request)`. I should add a test for that though...